doc/concepts/morphology.rst

```diff
@@ -3,94 +3,164 @@
    ``dist``,       :py:class:`point <arbor.mpoint>`,   the center and radius of the distal end.
-   ``tag``,        integer, tag meta-data.
+   ``tag``,        integer (:term:`tag`),              "tag meta-data, can be used to classify segments of the same kind (ex: soma, dendrite, but also arbitrary use-defined groups"
+
+.. figure:: ../gen-images/term_segments.svg
+  :width: 300
+  :align: center
+
+  Four segments arranged head to toe.
+
+.. glossary::
+
+  branch
+    A branch is the longest possible unbranched sequence of :term:`segments <segment>`.
+
+.. figure:: ../gen-images/term_branch.svg
+  :width: 300
+  :align: center
+
+  A branch corresponding to the previous segments.
+
+.. glossary::
+
+  mlocation
+    A location is not a point in 3D space, but a point in the cable cell morphology's
+    coordinate system. It is defined by a specific branch and a position along the length of the branch.
+
+.. csv-table:: The properties of :term:`mlocation`.
+   :widths: 10, 10, 30
+   :align: center
 
-.. _morph-tag-definition:
+   **Field**,      **Type**,        **Description**
+   ``branch``,     integer,         id of the branch
+   ``pos``,        real (0≤pos≤1),  position along the length of the branch
 
-A *tag* is an integer label on every segment, which can be used to define disjoint
-regions on cells.
-The meaning of tag values are not fixed in Arbor, however we typically use tag values that correspond
-to SWC `structure identifiers <http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html>`_.
+.. glossary::
+
+  cable
+    A cable is a subset of a :term:`branch`, and is thus defined as between two :term:`locations <mlocation>` on a particular branch.
+
+.. figure:: ../gen-images/term_cable.svg
+  :width: 300
+  :align: center
+
+  A cable shown in black is defined as a section :term:`located <mlocation>` between 0.2 and 0.8 of the length of the previous branch.
+
+.. glossary::
+
+  tag
+    A tag is an integer label on every segment, which can be used to define disjoint regions on cells.
+    The meaning of tag values are not fixed in Arbor, however we typically use tag values that correspond
+    to SWC `structure identifiers <http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html>`_.
+
+With these definitions, we can create proper definitions for *morphology* and *segment tree*.
+
+.. note::
+
+  NEURON uses different nomenclature for segments and branches. What's called a *section* in NEURON corresponds to an Arbor :term:`branch`. Arbor uses truncated cones to represent branches and segments, Neuron uses a weighted average radius to create cylinders (see "trapezoidal integration" in the `Neuron documentation <https://www.neuron.yale.edu/neuron/static/py_doc/modelspec/programmatic/topology/geometry.html>`_.). The number of *segments* (`nseg`) that NEURON uses for control over discretisation corresponds to an Arbor :term:`control volume`.
```

Comment:
Can this be split across several lines? Also, I would prefer:
"what's called a *section* in NEURON most closely corresponds to a :term:`branch` in Arbor" 
and 
"The *segments* that NEURON uses for control over discretisation ( by assigning `nseg` segments per section) most closely correspond to :term:`control volumes <control volume>` in Arbor."
It's probably better not to claim that they correspond exactly.